flake.nix

```diff
@@ -13,57 +15,58 @@
   outputs = {
     self,
     nixpkgs,
-    flake-utils,
+    systems,
     rust-overlay,
     ...
   }: let
-    gitRev = self.rev or self.dirtyRev or null;
-  in
-    flake-utils.lib.eachDefaultSystem (system: let
-      pkgs = import nixpkgs {
+    inherit (nixpkgs) lib;
+    eachSystem = lib.genAttrs (import systems);
```

Comment:
Dropping flake-utils would be nice but this is also slotting in a new dependency, albeit a small one https://github.com/nix-systems/default/blob/main/default.nix

Can we drop flake-utils and only depend on things in nixpkgs for the systems? I believe this helper can be written as `lib.genAttrs lib.systems.flakeExposed`

Reply:
@the-mikedavis You should bring back the `nix-systems` pattern. If your really don't like having it be a remote URL, use the `path:./flake.systems.nix` style instead, with `flake = false`. That ensures that if someone wants, they can `inputs.helix.inputs.systems.follows =` some other list of systems, and build for some unsupported platform without a warranty.